lib/galaxy/model/__init__.py

```diff
@@ -1872,16 +1872,22 @@ def set_extra_files_path(self, extra_files_path):
             self.external_extra_files_path = extra_files_path
     extra_files_path = property(get_extra_files_path, set_extra_files_path)
 
-    def _calculate_size(self):
+    def _calculate_size(self, wait_seconds=0):
         if self.external_filename:
+            exists = False
+            while exists is False and wait_seconds > 0:
+                exists = os.path.exists(self.external_filename)
+                if exists is False:
+                    time.sleep(0.25)
+                    wait_seconds -= 0.25
```

Comment:
Since that's an external file (so it should be there before it is added to Galaxy), maybe that's not needed?